pkg/controller/connectivitycheck/connectivity_check_controller.go

```diff
@@ -0,0 +1,404 @@
+				results = append(results, endpointInfo{
+					hostName: address.IP,
+					port:     strconv.Itoa(int(port.Port)),
+					nodeName: *address.NodeName,
+				})
+			}
+		}
+	}
+	return results, nil
+}
+
+func (c *connectivityCheckTemplateProvider) getTemplatesForAPILoadBalancerChecks(recorder events.Recorder) []*v1alpha1.PodNetworkConnectivityCheck {
+	var templates []*v1alpha1.PodNetworkConnectivityCheck
+	infrastructure, err := c.infrastructureLister.Get("cluster")
+	if err != nil {
+		recorder.Warningf("EndpointDetectionFailure", "error detecting api load balancer endpoints: %v", err)
+		return nil
+	}
+
+	apiUrl, err := url.Parse(infrastructure.Status.APIServerURL)
+	if err != nil {
+		recorder.Warningf("EndpointDetectionFailure", "error detecting external api load balancer endpoint: %v", err)
+
+	} else {
+		templates = append(templates, NewPodNetworkConnectivityCheckTemplate(apiUrl.Host, "openshift-network-diagnostics", withTarget("load-balancer", "api-external")))
+	}
+
+	apiInternalUrl, err := url.Parse(infrastructure.Status.APIServerInternalURL)
+	if err != nil {
+		recorder.Warningf("EndpointDetectionFailure", "error detecting internal api load balancer endpoint: %v", err)
+	} else {
+		templates = append(templates, NewPodNetworkConnectivityCheckTemplate(apiInternalUrl.Host, "openshift-network-diagnostics", withTarget("load-balancer", "api-internal")))
+	}
+	return templates
+}
+
+type endpointInfo struct {
+	hostName string
+	port     string
+	nodeName string
+}
+
+func withTarget(label, target string) func(check *v1alpha1.PodNetworkConnectivityCheck) {
+	return WithTarget(label + "-" + target)
+}
+
+func Start(ctx context.Context, kubeConfig *rest.Config) error {
+	protoKubeConfig := rest.CopyConfig(kubeConfig)
+	protoKubeConfig.ContentType = "application/vnd.kubernetes.protobuf,application/json"
+	eventRecorder := &eventrecorder.LoggingRecorder{}
+	kubeClient, err := kubernetes.NewForConfig(protoKubeConfig)
+	if err != nil {
+		return err
+	}
+	configClient, err := configv1client.NewForConfig(kubeConfig)
+	if err != nil {
+		return err
+	}
+	operatorClient, dynamicInformers, err := genericoperatorclient.NewClusterScopedOperatorClient(kubeConfig, operatorv1.GroupVersion.WithResource("openshiftapiservers"))
+	if err != nil {
+		return err
+	}
+	operatorcontrolplaneClient, err := operatorcontrolplaneclient.NewForConfig(kubeConfig)
+	if err != nil {
+		return err
+	}
+	apiextensionsClient, err := apiextensionsclient.NewForConfig(kubeConfig)
+	if err != nil {
+		return err
+	}
+	apiextensionsInformers := apiextensionsinformers.NewSharedInformerFactory(apiextensionsClient, 10*time.Minute)
+	kubeInformersForNamespaces := v1helpers.NewKubeInformersForNamespaces(kubeClient,
+		"",
+		"openshift-config",
+		"openshift-config-managed",
+		"openshift-cluster-network-operator",
+		"openshift-network-diagnostics",
+		metav1.NamespaceSystem,
+		"openshift-kube-apiserver",
+		"openshift-apiserver",
+		"openshift-oauth-apiserver",
```

Comment:
This set of namespaces seems unnecessarily large

Reply:
done